cax/main.py

```diff
@@ -288,10 +292,48 @@ def massive():
         docs = list(collection.find(query,
                                     sort=sort_key,
                                     projection=['start', 'number','name',
-                                                'detector', '_id']))
+                                                'detector', '_id', 'processor']))
 
         for doc in docs:
+            
+            #If no_batch_mode is chosen, no jobs will submitted to batch queues
+            if no_batch_mode == True:
+                
+                #Define basic command:
+                basic_command = """#!/bin/bash
+cax --once {config} --name {name}
+"""
+                
+                print("no-batch-mode")
+                
+                #Pre-read the config file for the no-batch-mode modus:
+                config.set_json(os.path.abspath(args.config_file))
+                task_list = config.get_config(config.get_hostname())['task_list']
+                
+                #This is a list of tasks which refer to apply corrections to a run:
+                task_list_correction = ['AddElectronLifetime', 'AddGains', 'AddDriftVelocity', 'SetS2xyMap', 'SetLightCollectionEfficiency', 'SetFieldDistortion', 'SetNeuralNetwork']
+                
+                #--------------------
+                #Start with single activities: Apply corrections only to data sets which do not have any correction yet:
+                if bool(set(task_list) & set(task_list_correction)) or task_list_correction == task_list:
```

Comment:
I'm not so familiar with this bitwise operation on Python sets, but isn't the second condition after the ```or``` included in the first condition (i.e. don't need the ```or```)?

Also, what if I want to run in no-batch mode without running corrections?